src/semafour.cc

```diff
@@ -257,6 +260,26 @@ void Semafour::Close(const v8::FunctionCallbackInfo<v8::Value>& args) {
 }
 
 
+void Semafour::TryWait(const v8::FunctionCallbackInfo<v8::Value>& args) {
+  Semafour* sem = UNWRAP(Semafour);
+  Isolate* isolate = args.GetIsolate();
+  int r;
+
+  // Handle synchronous case.
+  r = sem->Wait(true);
+
+  if (r != 0 && r != -EAGAIN) {
```

Comment:
This won't be negative EAGAIN will it? I though it would just be EAGAIN.

Reply:
Yes it will, because we negate errno when we return it, regardless of whether it is EAGAIN or not.